freqtrade/freqtradebot.py

```diff
@@ -443,6 +451,53 @@ def create_trade(self, pair: str) -> bool:
         else:
             return False
 
+#
+# BUY / increase positions / DCA logic and methods
+#
+    def process_open_trade_positions(self):
+        """
+        Tries to execute additional buy or sell orders for open trades (positions)
+        """
+        # Walk through each pair and check if it needs changes
+        for trade in Trade.get_open_trades():
+            try:
+                self.adjust_trade_position(trade)
+            except DependencyException as exception:
+                logger.warning('Unable to adjust position of trade for %s: %s',
+                               trade.pair, exception)
+
+    def adjust_trade_position(self, trade: Trade):
+        """
+        Check the implemented trading strategy for adjustment command.
+        If the strategy triggers the adjustment, a new order gets issued.
+        Once that completes, the existing trade is modified to match new data.
+        """
+        # If there is any open orders, wait for them to finish.
+        for order in trade.orders:
+            if order.ft_is_open:
+                return
+
+        logger.debug(f"adjust_trade_position for pair {trade.pair}")
+        current_rate = self.exchange.get_rate(trade.pair, refresh=True, side="buy")
+        current_profit = trade.calc_profit_ratio(current_rate)
+        stake_amount = strategy_safe_wrapper(self.strategy.adjust_trade_position,
+                                             default_retval=None)(
+            pair=trade.pair, trade=trade, current_time=datetime.now(timezone.utc),
+            current_rate=current_rate, current_profit=current_profit)
+
+        if stake_amount is not None and stake_amount > 0.0:
+            # We should increase our position
+            self.execute_entry(trade.pair, stake_amount, trade=trade)
+
+        if stake_amount is not None and stake_amount < 0.0:
+            # We should decrease our position
+            # TODO: Selling part of the trade not implemented yet.
+            logger.error(f"Unable to decrease trade position / sell partially"
+                         f" for pair {trade.pair}, feature not implemented.")
+            return
+
+        return
```

Comment:
These 2 statements are not really necessary (python will return None implicitly if there's no return statement present).